connection_manager.go

```diff
@@ -206,6 +256,10 @@ func (cm *connectionManager) createConnection() (*connection, error) {
 }
 
 func (cm *connectionManager) get() (*connection, error) {
+	if cm.connectionCounter.isLessThan(cm.minConnections) {
+		return cm.create()
+	}
```

Comment:
That's exactly what would happen. Let me improve this. Aren't code reviews grand? :smile: